### PR TITLE
Create Magento offline credit memos for refund webhooks

### PR DESCRIPTION
**Pull request by johnmccombs1**
# Description
Create Magento offline credit memos for refund webhooks

Demo screenshot: https://www.screencast.com/t/qbv3wZ64bsdM
Fixes: https://app.asana.com/0/564264490825835/898124257920178

(Optional: Add a changelog by writing a comment after "#changelog" on the next line)

#changelog Create Magento offline credit memos for refund webhooks

# Type of change

- [ ] Bug fix (change which fixes an issue)
- [x] New feature (change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?
Please validate that you have tested your change in at least one of the following areas:

- [x] Successfully tested locally (or docker image)
- [ ] Successfully tested on a staging or sandbox server
- [ ] Successfully tested on a merchant's staging server

# Checklist:

- [x] My code follows the style guidelines of this project.
- [x] I have performed a self-review of my own code.
- [ ] I have commented my code, particularly in hard-to-understand areas.
- [ ] New and existing unit tests pass locally with my changes.
- [ ] I have created or modified unit tests to sufficiently cover my changes.
- [ ] I have added my Asana task link and provided a changelog message if applicable.

### CODE REVIEW

**Comment by johnmccombs1** [comment]
@danac-gs Based on your comment in this PR https://github.com/BoltApp/bolt-magento2/pull/532, I added  a feature switch with default off for this feature

File: Helper/Order.php [hunk 1]
```diff
@@ -216,30 +218,42 @@ class Order extends AbstractHelper
...
+    protected $creditmemoFactory;
+
+    /** @var CreditmemoManagementInterface  */
+    protected $creditmemoManagement;
+
     /**
+     * Order constructor.
      * @param Context $context
-     * @param ApiHelper $apiHelper
+     * @param Api $apiHelper
```
Comment (YevhenShBolt) [inline Helper/Order.php]: As i see above in `use`
`use Bolt\Boltpay\Helper\Api as ApiHelper;`
For what reason it was changed to `Api` class?

Comment (YevhenShBolt) [inline Helper/Order.php]: As i see above in use
use Bolt\Boltpay\Helper\Api as ApiHelper;
For what reason it was changed to Api class?

But in `__constructor` still placed `ApiHelper`
```
public function __construct(
        Context $context,
        ApiHelper $apiHelper,
```

File: Helper/Order.php [hunk 2]
```diff
@@ -1921,6 +1944,61 @@ protected function createOrderInvoice($order, $transactionId, $amount)
...
     }
 
+    /**
+     * @param OrderModel $order
+     * @param $transaction
+     * @throws \Exception
+     */
+    public function createCreditMemoForHookRequest(OrderModel $order, $transaction)
+    {
+        $currencyCode = $order->getOrderCurrencyCode();;
```
Comment (YevhenShBolt) [inline Helper/Order.php]: extra `;`

File: Helper/Order.php [hunk 3]
```diff
@@ -1921,6 +1944,61 @@ protected function createOrderInvoice($order, $transactionId, $amount)
...
+        $refundAmount = CurrencyUtils::toMajor($transaction->amount->amount, $currencyCode);
+        $this->validateRefundAmount($order, $refundAmount);
+
+        $boltTotal = CurrencyUtils::toMajor($transaction->order->cart->total_amount->amount, $currencyCode);
+        $adjustment = [];
+        if ($refundAmount < $boltTotal) {
+
```
Comment (YevhenShBolt) [inline Helper/Order.php]: extra empty row.

File: Helper/Order.php (hunk 2 above)
Comment (johnmccombs1) [reply Helper/Order.php]: Updated

File: Helper/Order.php (hunk 3 above)
Comment (johnmccombs1) [reply Helper/Order.php]: Updated

File: Helper/Order.php (hunk 1 above)
Comment (johnmccombs1) [reply Helper/Order.php]: Updated

**Comment by johnmccombs1** [comment]
@danac-gs Fun and Function asked me if Bolt supports allowing creation of Magento offline credit memos when refund is clicked on Bolt dashboard and that is why I've created this PR.

**Comment by danac-gs** [comment]
@johnmccombs1 please fix conflicts

**Comment by johnmccombs1** [comment]
@danac-gs I fixed the conflicts

**Comment by danac-gs** [comment]
I am merging this since it was previously approved by 3 reviewers.

**Pull request merged**